server/xpub-client/pubsub.js

```diff
@@ -0,0 +1,33 @@
+class PubsubClient {
+  constructor(PubsubManager) {
+    this.manager = PubsubManager
+    this.pubsub = null
+    this.publishingInterval = null
+  }
+
+  async initialize() {
+    this.pubsub = await this.manager.getPubsub()
+  }
+
+  publish(message, dataCallback) {
+    this.pubsub.publish(message, dataCallback())
```

Comment:
Could `dataCallback` be changed to a value here and in the `stopPublishingOnInterval` ? The reason `startPublishingOnInterval` is passed a callback instead of a value is that the value is likely to change each time the interval is called ie: progress. The first time this is called, progress may be 0 but you'll want this value to change over time. Publish (and the extra message called in `stopPublishingOnInterval` ) are one off calls so the value will not change.

Reply:
This can be raised as a new PR later